sqlx-core/src/mssql/error.rs

```diff
@@ -49,4 +49,30 @@ impl DatabaseError for MssqlDatabaseError {
     fn into_error(self: Box<Self>) -> Box<dyn StdError + Send + Sync + 'static> {
         self
     }
+
+    fn kind(&self) -> Option<ErrorKind> {
+        match self.0.number {
+            error_codes::UNIQUE_VIOLATION => Some(ErrorKind::UniqueViolation),
+            error_codes::NOT_NULL_VIOLATION => Some(ErrorKind::NotNullViolation),
+            error_codes::CHECK_FOREIGN_KEY_VIOLATION => {
+                let constraint = self.message().splitn(7, char::is_whitespace).nth(6)?;
+                match &constraint[..11] {
+                    "FOREIGN KEY" => Some(ErrorKind::ForeignKeyViolation),
+                    "CHECK const" => Some(ErrorKind::CheckViolation),
+                    _ => None,
+                }
+            }
```

Comment:
@abonander although the implementation for MSSQL works fine, do you think the complexity is worth it?